core/imt/src/RArena.hxx

```diff
@@ -0,0 +1,20 @@
+#include "tbb/task_arena.h"
+namespace ROOT {
+    namespace Internal {
+      // Wrapper for tbb::task_arena.
+      //
+      //tbb::task_arena is an alias of tbb::interface7::task_arena, which doesn't allow
+      // to forward declare tbb::task_arena. To avoid code breaking on tbb interface changes
+      // we don't forward declare tbb::interface7::task_arena and instead we wrap tbb::task_arena
+      // in the forward declared class RArena
+      class RArena {
+      public:
+        /// Access the wrapped object and allow to call its methods.
+        tbb::task_arena *operator->() {
+            return fTBBArena.get();
+        }
+      private:
+         std::unique_ptr<tbb::task_arena> fTBBArena{new tbb::task_arena()};
```

Comment:
I think so! Will check

Reply:
on a second thought... isn't operator->() required to yield a ptr result?

Reply:
Ok, not really. But then tbb:task_arena should overload the operator -> as well. See: https://stackoverflow.com/a/20688236

Reply:
I was actually a little bit confused about the overloading of that operator and the invisible application of another -> 😅

Reply:
you can just return `&fTBBArena` from `operator->`, that's not a problem

Reply:
true lol

Reply:
Then, what would you do with the name of the class? Because it keeps an arena as data member, but the only way to access it is through the pointer returned by the operator ->

Reply:
what _is_ this class? the only API it offers is an `operator->` that lets you operate on a `task_arena`, but it does not have pointer semantics anymore because copying a `RArenaPtr` actually duplicates the `task_arena`..? So maybe a getter method instead of `operator->` would be a better interface..?

Reply:
why do you even need an accessor for the member? You give full access to the member anyway, so there is no encapsulation. Just have it as a public member or inherit it:
```
struct RArenaPtr : tbb:task_arena{};
```